great_expectations/datasource/fluent/postgres_datasource.py

```diff
@@ -16,4 +19,4 @@ class PostgresDatasource(SQLDatasource):
     """
 
     type: Literal["postgres"] = "postgres"  # type: ignore[assignment]
-    connection_string: PostgresDsn
+    connection_string: Union[ConfigStr, PostgresDsn]
```

Comment:
Does ConfigStr fail validation if there is no variables present?

Reply:
Yes, exactly. If it's not a valid template string the type cannot be created.